sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -291,45 +294,58 @@ def call(self):
 
     side_input_values = [self._side_input_values[side_input]
                          for side_input in self._applied_ptransform.side_inputs]
-
     try:
-      evaluator = self._transform_evaluator_registry.get_evaluator(
-          self._applied_ptransform, self._input_bundle,
-          side_input_values, scoped_metrics_container)
-
-      if self._fired_timers:
-        for timer_firing in self._fired_timers:
-          evaluator.process_timer_wrapper(timer_firing)
-
-      if self._input_bundle:
-        for value in self._input_bundle.get_elements_iterable():
-          evaluator.process_element(value)
+      self.attempt_call(metrics_container,
+                        scoped_metrics_container,
+                        side_input_values)
+    except Exception as e:
```

Comment:
<!--new_thread; commit:eb2acfa855db99ab4b826344656ec29dcdb175e3; resolved:0-->
Can you add logging of the traceback every time it fails?  (see `traceback.format_exc()`).